### PR TITLE
Add core rewards spec tests

### PR DESCRIPTION
**Pull request by terencechain**
Review and merge #5965 first

This PR implements the newly introduce core rewards tests:
https://github.com/ethereum/eth2.0-spec-tests/tree/master/tests/mainnet/phase0/rewards/core/pyspec_tests

Both mainnet and minimal are passing as of commit `ddcf607`

Special thanks to @protolambda for some of the help on getting the test to pass

### CODE REVIEW

File: beacon-chain/core/epoch/precompute/new.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func New(ctx context.Context, state *stateTrie.BeaconState) ([]*Validator, *Bala
 
 	if err := state.ReadFromEveryValidator(func(idx int, val *stateTrie.ReadOnlyValidator) error {
 		// Was validator withdrawable or slashed
-		withdrawable := currentEpoch >= val.WithdrawableEpoch()
+		withdrawable := prevEpoch+1 >= val.WithdrawableEpoch()
```
Comment (nisdas) [inline beacon-chain/core/epoch/precompute/new.go:30]: why the change ?

Comment (terencechain) [reply beacon-chain/core/epoch/precompute/new.go:30]: To align with the spec. It shouldn't matter since we don't process rewards and penalties until second epoch. See more discussion: https://github.com/ethereum/eth2.0-specs/issues/1846

Either works but I decided to change it to `prevEpoch+1` so it reads exactly the same as the spec, the new comers wont be as confused when they read it

File: beacon-chain/core/epoch/precompute/attestation.go [hunk 2]
```diff
@@ -182,5 +183,28 @@ func UpdateBalance(vp []*Validator, bBal *Balance) *Balance {
 			}
 		}
 	}
+
+	if params.BeaconConfig().EffectiveBalanceIncrement > bBal.ActiveCurrentEpoch {
```
Comment (nisdas) [inline beacon-chain/core/epoch/precompute/attestation.go:187]: can we have a test case specifically for all these conditions ?

Comment (rauljordan) [reply beacon-chain/core/epoch/precompute/attestation.go:187]: Clicked the merge too soon @nisdas forgot. there was no branch protection again ugh, we. can perhaps do in a separate PR

Comment (nisdas) [reply beacon-chain/core/epoch/precompute/attestation.go:187]: No prob, we can maybe add it in a separate PR.

Comment (terencechain) [reply beacon-chain/core/epoch/precompute/attestation.go:187]: All good guys! I clicked pushed an hour ago and just realized now the pr has been merged.
Here's the follow up: https://github.com/prysmaticlabs/prysm/pull/6007
